docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,331 @@
+
+In these systems, it’s sensible to split the verification so as each library is
+operated on an individual basis.
+
+If you encounter timeouts when trying to verify the main entry point contract to
+the system, check the impact of the libraries on the verification by summarizing
+all external library (delegate) calls as `NONDET`, using the option
+`summarizeExtLibraryCallsAsNonDetPreLinking` as follows:
+
+```
+certoraRun ... --prover_args '-summarizeExtLibraryCallsAsNonDetPreLinking true'
+```
+
+```{note}
+This option is only applied for _external_ library calls, or `delegatecall`s.
+Internal calls are automatically inlined by the Solidity compiler and are 
+subject to summarizations specified in the spec file's `methods` block.
+```
+
+Alternatively, if you wish to apply a "catch-all" summary for all the methods of
+a specific library, you can write in the methods block of the spec:
+
+```
+methods {
+    function MyBigLibrary._ external => NONDET;
+    function MyBigLibrary._ internal => NONDET;
+}
+```
+
+The above snippet has the effect of summarizing as `NONDET` all external calls
+to the library and _internal_ ones as well. All summary types except ghost
+summaries can be applied. 
+
+## Simplifying the Source Code 
+% aka  Munging / Verifiable-Code-Antipatterns
+
+Simplifying the source code of the program under verification can be a valuable last resort for obtaining useful verification results.
+In the following, we describe some code patterns that have proven to be very difficult for the Prover and thus are good targets for code simplification.
+Note that the occurrence of these patterns is not always a problem, so they should be looked at in conjunction with the difficulty statistics and generally a holistic view of the program under verification.
+
+### Passing Complex Structs
+
+A common culprit for high memory complexity are complex datastructures that are
+passed from the specification to the program, or also inside the program.
+Especially problematic are `struct` types that contain many dynamically-sized
+arrays. 
+
+% TODO: which calls exactly? external calls? all of them?
+
+```cvl
+rule myRule() {
+    MyStruct x;
+    foo(x);
+
+}
+```
+
+```solidity
+struct MyStruct {
+    // several dynamically-sized arrays
+    bytes b;
+    string s;
+    uint[] u1;
+    uint8[] u2;
+}
+
+function foo(MyStruct x) public {
+    ...
+}
+```
+
+### Memory and Storage in Inline Assembly
+
+% Shows through storage/memory analysis failures ("Global Problems" pane).
+
+% Q: could we have "background" boxes or so -- indicating information that helps with a deeper understanding, but does only indirectly relate to using the tool -- just some place for rambling :-) 
+
+```{note}
+Background: The Certora Prover works on EVM bytecode as its input. To the
+bytecode, the address space of both Storage and Memory are flat number lines.
+That two contract fields `x` and `y` don't share the same memory an arithmetic
```

Comment:
The `That` here is odd.

Reply:
right .. there was an `is` missing later in that sentence ..
(naturally, this section is of the "background" category ... not sure about it ..)